src/components/CategoryInput.tsx

```diff
@@ -23,23 +19,33 @@ const CategoryInput = ({ setSelectedCategoryInfo }: CategoryInputProps) => {
     fetchAndStoreAllCategories();
   }, []);
 
+  const handleChangeCategory = (categoryName: string) => {
+    if (categoryName === "none") {
+      setSelectedCategoryInfo({
+        id: -1,
+        name: "",
+      });
+    } else {
+      const categoryInfo: ICategoryInfo = allCategories.filter(
+        (catInfo: ICategoryInfo) => catInfo.name === categoryName
+      )[0];
+      setSelectedCategoryInfo({
+        id: categoryInfo.id,
+        name: categoryInfo.name,
+      });
```

Comment:
if `ICategoryInfo` matches with the signature of setSelectedCategoryInfo, then would it make sense to just?

```suggestion
      setSelectedCategoryInfo(categoryInfo);
```

Reply:
lol yes, you're right. I've made that change now